File: dotnet/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example3/SoftwareDeveloper.cs
namespace CompositeDesignPattern.Example3;

public class SoftwareDeveloper : Employee
{
	public SoftwareDeveloper(string name, double salary) : base(name, salary)
	{
	}

	public override void Print()
	{
		Console.WriteLine($"\tSoftware Developer Name\t\t: {GetName()}");
		Console.WriteLine($"\tSoftware Developer Salary\t: {GetSalary()}");
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example3/src/SoftwareDeveloper.ts
import { Employee } from './Employee';

export class SoftwareDeveloper extends Employee {
  constructor(name: string, salary: number) { super(name, salary); }

  print(): void {
    console.log(`\tSoftware Developer Name\t\t: ${this.getName()}`);
    console.log(`\tSoftware Developer Salary\t: ${this.getSalary()}`);
  }
}
